marklogic-data-hub-central/ui/src/components/flows/flows.tsx

```diff
@@ -554,6 +572,80 @@ const Flows: React.FC<Props> = ({
     }
   };
 
+  const getLocalStorageDataUser = () => {
+
+    if (getUserPreferencesLS()) {
+      if (getUserPreferencesLS()?.selectedStepsDataUser?.selectedStepOptions) {
+        setSelectedStepOptions(getUserPreferencesLS().selectedStepsDataUser.selectedStepOptions);
+      }
+      if (getUserPreferencesLS()?.selectedStepsDataUser?.arrayLoadChecksSteps) {
+        setArrayLoadChecksSteps(getUserPreferencesLS().selectedStepsDataUser.arrayLoadChecksSteps);
+      }
+      if (getUserPreferencesLS()?.selectedStepsDataUser?.selectedStepDetails) {
+        setSelectedStepDetails(getUserPreferencesLS().selectedStepsDataUser.selectedStepDetails);
+      }
+    }
+  };
+
+  const tryParseJson = (userPreferences) => {
+    try {
+      return JSON.parse(userPreferences);
+    } catch (e) {
+      return false;
+    }
+  };
+
+  const getUserPreferencesLS = () => {
+    let userPreferences, oldOptions;
+    const sessionUser = localStorage.getItem("dataHubUser");
+
+    if (sessionUser) {
+      userPreferences = getUserPreferences(sessionUser);
+      if (userPreferences) {
+        oldOptions = tryParseJson(userPreferences);
+
+        //if (Object.keys(oldOptions.selectedStepsDataUser?.selectedStepOptions).length === 0) return false
+      } else { return false; }
+    }
+
+    return oldOptions;
+  };
+
+  const saveLocalStoragePreferences = (value: boolean, saveSelectedStepsDataUser?: boolean) => {
+
+    const sessionUser = localStorage.getItem("dataHubUser");
+    if (getUserPreferencesLS()) {
```

Comment:
Again the same suggestion. Let's avoid fetching the userPreferences multiple times. Try fetching them once at the beginning of the current scope and then use the return value further as needed, within the same scope.